modules/unsupported/mongodb/src/test/java/org/geotools/data/mongodb/MongoFeatureSourceTest.java

```diff
@@ -0,0 +1,142 @@
+
+      SimpleFeatureCollection features = source.getFeatures(q);
+      SimpleFeatureIterator it = features.features();
+      try {
+          assertTrue(it.hasNext());
+          assertFeature(it.next(), 1);
+      }
+      finally {
+          it.close();
+      }
+    }
+
+    public void testEqualToFilter() throws Exception {
+        FilterFactory2 ff = CommonFactoryFinder.getFilterFactory2();
+        PropertyIsEqualTo f = ff.equals(ff.property("properties.stringProperty"), ff.literal("two"));
+
+        SimpleFeatureSource source = dataStore.getFeatureSource("ft1");
+        Query q = new Query("ft1", f);
+        
+        assertEquals(1, source.getCount(q));
+        assertEquals(new ReferencedEnvelope(2d,2d,2d,2d,DefaultGeographicCRS.WGS84), source.getBounds(q));
+
+        SimpleFeatureCollection features = source.getFeatures(q);
+        SimpleFeatureIterator it = features.features();
+        try {
+            assertTrue(it.hasNext());
+            assertFeature(it.next(), 2);
+        }
+        finally {
+            it.close();
+        }
+    }
+
+    public void testLikeFilter() throws Exception {
+        FilterFactory2 ff = CommonFactoryFinder.getFilterFactory2();
+        PropertyIsLike f = ff.like(ff.property("properties.stringProperty"), "on%", "%", "_", "\\");
+
+        SimpleFeatureSource source = dataStore.getFeatureSource("ft1");
+        Query q = new Query("ft1", f);
+
+        assertEquals(1, source.getCount(q));
+        assertEquals(new ReferencedEnvelope(1d,1d,1d,1d,DefaultGeographicCRS.WGS84), source.getBounds(q));
+
+        SimpleFeatureCollection features = source.getFeatures(q);
+        SimpleFeatureIterator it = features.features();
+        try {
+            assertTrue(it.hasNext());
+            assertFeature(it.next(), 1);
+        }
+        finally {
+            it.close();
+        }
+
+        // check full string match
+        f = ff.like(ff.property("properties.stringProperty"), "n%", "%", "_", "\\");
+
+        source = dataStore.getFeatureSource("ft1");
+        q = new Query("ft1", f);
+
+        // no feature should match
+        assertEquals(0, source.getCount(q));
+    }
+
+    public void testLikePostFilter() throws Exception {
+        FilterFactory2 ff = CommonFactoryFinder.getFilterFactory2();
+        // wrapping the property name in a function that is not declared as
+        // supported in the filter capabilities (i.e. Concatenate) will make the
+        // filter a post-filter
+        PropertyIsLike f = ff.like(ff.function("Concatenate",
+                ff.property("properties.stringProperty"), ff.literal("test")),
+                "on%", "%", "_", "\\");
+
+        SimpleFeatureSource source = dataStore.getFeatureSource("ft1");
+        Query q = new Query("ft1", f, new String[] { "geometry" });
+
+        // filter should match just one feature
+        assertEquals(1, source.getFeatures(q).size());
```

Comment:
Thanks @ridethepenguin (and @tbarsballe for checking) - this was the test I was looking for.